pkg/registry/generic/rest/streamer.go

```diff
@@ -56,6 +59,16 @@ func (s *LocationStreamer) InputStream(apiVersion, acceptHeader string) (stream
 	if err != nil {
 		return
 	}
+
+	if resp.StatusCode == http.StatusInternalServerError {
+		bodyBytes, err := ioutil.ReadAll(resp.Body)
```

Comment:
Well... That's basically what @feihujiang has written here, and it has this downside. I guess what you're saying that one should detect the status code and return an error if it's anything but 200, yet still return a valid stream. That's counter-intuitive--usually if a go function returns an error one expects to ignore the rest of the return parameters; in this case we'd still require them to close the body.